integration-tests/src/test/resources/sitconfig/java/SitConfigTests.java

```diff
@@ -0,0 +1,554 @@
+import java.sql.Statement;
+import java.util.*;
+import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.management.MBeanServerConnection;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+import weblogic.diagnostics.descriptor.WLDFHarvestedTypeBean;
+import weblogic.diagnostics.descriptor.WLDFInstrumentationMonitorBean;
+import weblogic.diagnostics.descriptor.WLDFResourceBean;
+import weblogic.j2ee.descriptor.wl.JDBCConnectionPoolParamsBean;
+import weblogic.j2ee.descriptor.wl.JDBCDataSourceBean;
+import weblogic.j2ee.descriptor.wl.JDBCDriverParamsBean;
+import weblogic.j2ee.descriptor.wl.JMSBean;
+import weblogic.j2ee.descriptor.wl.UniformDistributedTopicBean;
+import weblogic.management.ManagementException;
+import weblogic.management.configuration.*;
+import weblogic.management.jmx.MBeanServerInvocationHandler;
+import weblogic.management.mbeanservers.domainruntime.DomainRuntimeServiceMBean;
+import weblogic.management.mbeanservers.edit.ConfigurationManagerMBean;
+import weblogic.management.mbeanservers.edit.EditServiceMBean;
+import weblogic.management.mbeanservers.runtime.RuntimeServiceMBean;
+import weblogic.management.runtime.DomainRuntimeMBean;
+import weblogic.management.runtime.ServerRuntimeMBean;
+
+public class SitConfigTests {
+
+  private MBeanServerConnection domainRuntimeMbs, editMbs, runtimeMbs;
+  private JMXConnector jmxConnector;
+  private static ObjectName service;
+  private DomainRuntimeMBean domainRuntimeMBean;
+  private DomainRuntimeServiceMBean domainServiceMBean;
+  private RuntimeServiceMBean runtimeServiceMBean;
+  private ServerRuntimeMBean serverRuntime;
+  private EditServiceMBean editServiceMBean;
+
+  private DomainMBean editdomain, editTree;
+  private ConfigurationManagerMBean cfgMgr;
+  private static final String JNDI = "/jndi/";
+
+  private final String adminHost;
+  private final String adminPort;
+  private final String adminUser;
+  private final String adminPassword;
+
+  public static void main(String args[]) throws Exception {
+    String adminHost = args[0];
+    String adminPort = args[1];
+    String adminUser = args[2];
+    String adminPassword = args[3];
+    String testName = args[4];
+
+    SitConfigTests test = new SitConfigTests(adminHost, adminPort, adminUser, adminPassword);
+    // SitConfigTests test = new SitConfigTests("slc11vjg", "30051", "weblogic", "welcome1");
+
+    ServerRuntimeMBean runtimeMBean = test.runtimeServiceMBean.getServerRuntime();
+    println("Sitconfig State:" + runtimeMBean.isInSitConfigState());
+
+    if (testName.equals("testCustomSitConfigOverridesForDomain")) {
+      test.verifyDebugFlagJMXCore(true);
```

Comment:
What is the meaning of all of these values?  I'm especially interested in MaxMessageSize, RestartMax, and ConnectTimeout.  Are these values going to be tested later?  If so, we should pull them out as constants.

Reply:
Sorry for the confusion.  :).  I was asking for you to define these terms.  Instead, I was interested in the values.  For instance, why did you choose "7" as the value for harvestMaxCount?  Is this important to the test or just a value you hope will always be big enough?  I couldn't tell.

Reply:
Ryan -  Here we are not verifying the runtime behavior of the override attribute which is being tested in on-peremise testing effort.  Here we make sure the new value appears in the ConfigMBean Tree.

Reply:
These are just random parameters and values chosen, the main purpose of the test is to show/verify the overridden seen in the ConfigMBean tree.

Reply:
Added documentation in the test class about all these parameters and values.